addons/mail/static/src/models/discuss/discuss.js

```diff
@@ -184,6 +184,9 @@ function factory(dependencies) {
          */
         replyToMessage(message) {
             this.update({ replyingToMessage: [['link', message]] });
+            this.replyingToMessageOriginThreadComposer.update({
+                isLog: !message.is_discussion && !message.is_notification
+            });
```

Comment:
Would put a comment above for the intent, quite debatable and likely for changes in future (why not possible to change subtype?)

Reply:
Yeah the intent is not to prevent changing subtype, but to get the most likely expected subtype by default.

The really big problematic issue to avoid here is to reply to a colleague (who sent a note) and to send message to client by mistake instead.